tuf/api/metadata.py

```diff
@@ -348,10 +340,10 @@ def __init__(
         if version <= 0:
             raise ValueError(f"version must be > 0, got {version}")
         self.version = version
-        self.unrecognized_fields = unrecognized_fields or {}
+        self.unrecognized_fields: Mapping[str, Any] = unrecognized_fields or {}
```

Comment:
Interesting. I think this is the first time we've added an inline type annotation? Is this the only class property which is not set by an annotated constructor argument?

Reply:
The issue here is that `{}` has type "dict" so then mypy figures that our instance attribute type must be `Mapping[str, Any] | dict[str, Any]` which... does not break anything at all but also makes the whole idea of using Mapping a bit useless